src/vmm/src/rpc_interface.rs

```diff
@@ -811,6 +811,15 @@ impl RuntimeApiController {
         new_cfg: BlockDeviceUpdateConfig,
     ) -> Result<VmmData, VmmActionError> {
         let mut vmm = self.vmm.lock().expect("Poisoned lock");
+
+        // vhost-user-block updates
+        if new_cfg.path_on_host.is_none() && new_cfg.rate_limiter.is_none() {
```

Comment:
It will fail, because `update_block_device_path` or `update_block_rate_limiter` will be called, and they expect the device to be a virtio-block and not vhost-user-blk.

Reply:
Will that return a meaningful error to the user?

Reply:
It will fail with `MmioError::InvalidDeviceType` error, because `downcast_mut` will fail to cast device to the `VirtioBlock`

Reply:
I feel like we may need an integration test for that.

Reply:
added an integration test for this